src/routes/(app)/product/[id]/+page.server.ts

```diff
@@ -112,16 +113,32 @@ export const load = async ({ params, locals }) => {
 		.find({ productId: params.id })
 		.sort({ createdAt: 1 })
 		.toArray();
-	const subscriptions = await collections.paidSubscriptions
+	const freeProductSubscriptions = await collections.paidSubscriptions
 		.find({
 			...userQuery(userIdentifier(locals)),
-			paidUntil: { $gt: new Date() }
+			paidUntil: { $gt: new Date() },
+			[`freeProductsById.${product._id}`]: { $exists: true }
 		})
+		.sort({ [`freeProductsById.${product._id}.available`]: -1 })
+		.limit(1)
 		.toArray();
```

Comment:
you can remove this and just use a `.find()` from the results of the below db calls.

Eg  `subscriptions.find(s => (s.freeProductsById?.[product._id]?.available ?? 0) > 0)`

You should also sum the available amount from all the free product subscriptions.

Maybe you can just pass the available amount to the frontend instead of the whole subscriptions